x-pack/plugins/security_solution/cypress/README.md

```diff
@@ -208,6 +242,44 @@ Because of `cy.exec`, used to invoke `es_archiver`, it's necessary to override i
 
 > Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
 
+### CCS
+
+Tests running in CCS configuration need to care about two aspects:
+
+1. data (eg. to trigger alerts) is generated/loaded on the remote cluster
+2. queries (eg. detection rules) refer to remote indices
```

Comment:
By `data`, if we talk about detection engine, do you mean only source events here, or any data (including alerts, saved objects, etc), or something in between?

When a rule runs, it needs to query source events and write detection alerts. Sometimes, if we talk about "rules on alerts" feature, some rules called "building block rules" would query source events and write alerts which would be used by other rules to generate other alerts. So where would we store all of that in the CCS mode:

- initial source events
- "building block" alerts
- final alerts
- "building block" rules
- final rules

Is this flexible so we could decide where to keep which data per each test, or is it kinda fixed?

Reply:
@banderror when we talk about data, we are talking about all the initial setup you need to execute the tests, that could be of any kind, alerts, rules.... and so on.

In the automation we are working on, we are ingesting the data on the remote and querying from the source, since ingesting and querying the data from the same cluster would equal to a regular instance testing. I'm not sure if I'm following you about being flexible.

Reply:
@MadameSheema 

> In the automation we are working on, we are ingesting the data on the remote

So here by "data" do you only mean source events in `logs-*`, `metrics-*` etc indices? Or something else too?

> and querying from the source

Does that mean you create detection rules in the "local" cluster, and then these rules query source events in the "remote" cluster? In this case, rules would be Kibana Saved Objects stored in the "local" cluster.

What I was trying to say, in general users can have complex deployments of Elastic stack. E.g. a user could have "building block" rules and alerts stored in the "remote" cluster, and then they could have normal, final rules in the "local cluster". I don't know if there are plans to support these kind of complex setups from the product perspective, but I was thinking it's worth asking if the testing implementation for CCS allows to simulate and test such complex setups (this is what I meant by "flexible") where some data and saved objects could be in one cluster, and some - in another cluster.